cilium/cmd/encrypt_status.go

```diff
@@ -0,0 +1,144 @@
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// Cilium uses reqid 1 to tie the IPsec security policies to the matching state’
+	CiliumReqId = "1"
+)
+
+type void struct{}
+
+var (
+	voidType    void
+	countErrors int
+)
+
+var encryptStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Display the current encryption state",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium encrypt status")
+		getEncryptionMode()
+	},
+}
+
+func init() {
+	encryptCmd.AddCommand(encryptStatusCmd)
+	command.AddJSONOutput(encryptStatusCmd)
+}
+
+func getXfrmStats() (int, map[string]int) {
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		Fatalf("Cannot get a new proc FS: %s", err)
+	}
+	stats, err := fs.NewXfrmStat()
+	if err != nil {
+		Fatalf("Failed to read xfrm statistics: %s", err)
+	}
+	v := reflect.ValueOf(stats)
+	errorMap := make(map[string]int)
+	for i := 0; i < v.NumField(); i++ {
+		name := v.Type().Field(i).Name
+		value := v.Field(i).Interface().(int)
+		if value != 0 {
+			countErrors += value
+			errorMap[name] = value
+		}
+	}
+	return countErrors, errorMap
+}
+
+func countUniqueIPsecKeys() int {
+	// trying to mimic set type data structure
+	// using void data type as struct{} because it does not use any memory
+	keys := make(map[string]void)
+	if xfrmStates, err := netlink.XfrmStateList(netlink.FAMILY_ALL); err != nil {
+		Fatalf("Cannot get xfrm state: %s", err)
+	} else {
+		for _, v := range xfrmStates {
+			keys[string(v.Aead.Key)] = voidType
+		}
+	}
+	return len(keys)
+}
+
+func maxSequenceNumber() string {
+	maxSeqNum := ""
+	out, err := exec.Command("ip", "xfrm", "state", "list", "reqid", CiliumReqId).Output()
+	if err != nil {
+		Fatalf("Cannot get xfrm states: %s", err)
+	}
+	commandOutput := string(out)
+	lines := strings.Split(commandOutput, "\n")
+	regex := regexp.MustCompile("oseq[[:blank:]](0[xX][[:xdigit:]]+)?")
+	for _, line := range lines {
+		matched := regex.FindStringSubmatchIndex(line)
+		if matched != nil {
+			oseq := line[matched[2]:matched[3]]
+			if oseq > maxSeqNum {
+				maxSeqNum = oseq
+			}
+		}
+	}
+	return maxSeqNum
+}
+
+func getEncryptionMode() {
+	params := daemon.NewGetHealthzParamsWithTimeout(timeout)
+	params.SetBrief(&brief)
+	if resp, err := client.Daemon.GetHealthz(params); err != nil {
+		Fatalf("Cannot get daemon encryption status: %s", err)
+	} else {
+		encryptionStatusResponse := resp.Payload.Encryption
+		fmt.Printf("Encryption: %-26s\n", encryptionStatusResponse.Mode)
+		switch encryptionStatusResponse.Mode {
+		case models.EncryptionStatusModeIPsec:
+			dumpIPsecStatus()
+		}
+	}
+}
+
+func dumpIPsecStatus() {
+	keys := countUniqueIPsecKeys()
+	oseq := maxSequenceNumber()
+	fmt.Printf("Keys in use: %-26d\n", keys)
+	fmt.Printf("Max Seq. Number: %s/%s\n", oseq, "0xffffffff")
+	errCount, errMap := getXfrmStats()
+	fmt.Printf("Errors: %-26d\n", errCount)
+	if errCount != 0 {
+		for k, v := range errMap {
+			fmt.Printf("%s: %-26d\n", k, v)
```

Comment:
I think it would be good to indent those to clearly show they belong to `Errors:`:
```suggestion
			fmt.Printf("\t%s: %-26d\n", k, v)
```